cli/src/main/scala/com/codacy/analysis/cli/formatter/Sarif.scala

```diff
@@ -119,38 +124,62 @@ private[formatter] class Sarif(val stream: PrintStream, val executionDirectory:
         SarifReport.RuleProperties(category = issue.category.getOrElse(Pattern.Category.CodeStyle).toString))).toList
   }
 
-  private def createResults(filePath: Path,
-                            issues: Seq[Issue],
-                            artifacts: List[SarifReport.Artifact],
-                            rules: Seq[SarifReport.Rule]): Seq[SarifReport.Result] = {
-    val fileContents = executionDirectory./(filePath.toString).lines.toList
-    issues.map { issue =>
-      val message = SarifReport.Message(issue.message.text)
-      val filePath = s"$rootDirectory/${issue.filename.toString}"
-      val locations = List(
-        SarifReport.Location(SarifReport.PhysicalLocation(
-          artifactLocation = SarifReport
-            .ArtifactLocationIndexed(index = artifacts.indexWhere(_.location.uri == filePath), uri = filePath),
-          region = SarifReport.Region(startLine = issue.location.line))))
-
-      SarifReport.Result(
-        ruleIndex = rules.indexWhere(_.id == issue.patternId.value),
-        ruleId = issue.patternId.value,
-        message = message,
-        level = getSarifLevel(issue.level),
-        locations = locations,
-        partialFingerprints = SarifReport.PartialFingerprints("1", generatePrimaryLocationHash(issue, fileContents)))
-    }
+  private def createArtifacts(issues: Seq[Issue]): List[SarifReport.Artifact] = {
+    val filenames: Set[String] = issues.map(_.filename.toString)(collection.breakOut)
+    filenames.map(filename => SarifReport.Artifact(SarifReport.ArtifactLocation(s"$rootDirectory/$filename")))(
+      collection.breakOut)
+  }
+
+  private def createResults(
+    issues: Seq[Issue],
+    artifacts: List[SarifReport.Artifact],
+    rules: List[SarifReport.Rule],
+    getIssueSeverity: results.Result.Level => SarifReport.Level.Value): List[SarifReport.Result] = {
+
+    issues
```

Comment:
What do you think about moving this to a for comprehension and make this code more linear?